ibc-testkit/tests/core/ics02_client/update_client.rs

```diff
@@ -109,6 +109,81 @@ fn test_update_client_ok(fixture: Fixture) {
     );
 }
 
+#[rstest]
+// Tests successful submission of a header with a height below the latest
+// client's height and ensures that `ConsensusState` is stored at the correct
+// path (header height).
+fn test_update_client_with_prev_header() {
+    let client_id = ClientId::default();
+    let chain_id_b = ChainId::new("mockgaiaA-0").unwrap();
+    let latest_height = Height::new(0, 42).unwrap();
+    let height_1 = Height::new(0, 43).unwrap();
+    let height_2 = Height::new(0, 44).unwrap();
+
+    let mut ctx = MockContext::default().with_client_config(
+        MockClientConfig::builder()
+            .client_chain_id(chain_id_b.clone())
+            .client_type(tm_client_type())
+            .client_id(client_id.clone())
+            .latest_height(latest_height)
+            .build(),
+    );
+    let mut router = MockRouter::new_with_transfer();
```

Comment:
Small nit, but you can construct the `router` from the `fixture` function.

Reply:
The thing is: I need a `MockContext` with the Tendermint `ClientState` for this test, but the current fixture isn't configurable. Getting the router from fixture means constructing an unnecessary `MockContext`.

As an aside, this fixture should be overhauled in favor of a better approach like the one we have for connection tests.

Reply:
> As an aside, this fixture should be overhauled in favor of a better approach like the one we have for connection tests.

I'm trying to address this by defining a `mock_context` function like so:
```
fn mock_context(
    client_id: ClientId,
    height: Height,
    client_chain_id: Option<ChainId>,
    client_type: Option<ClientType>,
) -> MockContext {
    let client_config_builder = MockClientConfig::builder();

    if let Some(client_chain_id) = client_chain_id {
        client_config_builder.client_chain_id(client_chain_id);
    }

    if let Some(client_type) = client_type {
        client_config_builder.client_type(client_type);
    }

    let client_config = client_config_builder
        .client_id(client_id)
        .latest_height(height)
        .build();

   MockContext::default().with_client_config(client_config)
}
```

However, the compiler complains and prescribes calling `clone` on every line that modifies the `MockClientConfigBuilder`, which is obviously not ideal. Is there a better way to go about defining a configurable `mock_context` function?

I'm not sure where the connection tests are located.

Reply:
Looks like fully refactoring the client update tests to follow the approach that the connection tests use is a much bigger undertaking that deserves its own issues and PR. I will open a separate issue for that.

Reply:
Yes. Let's take care of this as a separate issue. Thank you Sean 🙏🏻